core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -104,22 +104,33 @@ public static WeekActivity createInstance(UserAnonymized userAnonymized, Goal go
 	public static WeekActivity createInstanceInactivity(UserAnonymized userAnonymized, Goal goal, ZonedDateTime startOfWeek)
 	{
 		WeekActivity result = createInstance(userAnonymized, goal, startOfWeek);
-		// add the days
-		// notice this doesn't take care of user time zone changes during the week
-		// so for consistency it is important that the batch script adding inactivity does so
-		for (int i = 0; i < 7; i++)
+		result.addInactivityDays();
+		return result;
+	}
+
+	public void addInactivityDays() 
+	{
+		// if the batch job has already run, skip
+		if(dayActivities.size() < 7)
 		{
-			ZonedDateTime startOfDay = startOfWeek.plusDays(i);
-			if (isInFuture(startOfDay, startOfWeek.getZone()))
+			// notice this doesn't take care of user time zone changes during the week
+			// so for consistency it is important that the batch script adding inactivity does so
+			for (int i = 0; i < 7; i++)
 			{
-				break;
-			}
-			if (goal.wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
-			{
-				result.addDayActivity(DayActivity.createInstanceInactivity(userAnonymized, goal, startOfDay));
+				ZonedDateTime startOfDay = getStartTime().plusDays(i);
+				if (isInFuture(startOfDay, getStartTime().getZone()))
+				{
+					break;
+				}
+				if (getGoal().wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
+				{
+					if(!dayActivities.stream().anyMatch(dayActivity -> dayActivity.getDate().getDayOfWeek().equals(startOfDay.getDayOfWeek())))
```

Comment:
Because you're suddenly adding inactivity via two ways: via the batch job (at the end of the week) and on-the-fly (at the start of the week).